src/main/kotlin/no/skatteetaten/aurora/boober/feature/ToxiproxySidecarSpec.kt

```diff
@@ -105,63 +130,101 @@ internal fun AuroraDeploymentSpec.proxynameDuplicateErrors() = getToxiproxyNames
+        }
+
+internal fun AuroraDeploymentSpec.serversAndPortsFromConfig(initialPort: Int): List<ToxiproxyConfig> {
+    return extractToxiproxyServersAndPorts()
+        .mapIndexed { i, (proxyname, serverVar, portVar, enabled) ->
+            val upstreamServer: String? = this.getOrNull("config/$serverVar")
+            val upstreamPort: String? = this.getOrNull("config/$portVar")
+
             ToxiproxyConfig(
                 name = proxyname,
                 listen = "0.0.0.0:${initialPort + i}",
-                upstream = uri.host + ":" + upstreamPort,
+                upstream = "$upstreamServer:$upstreamPort",
                 enabled = enabled
             )
-        } else null
-    }
-
-internal fun AuroraDeploymentSpec.serversAndPortsFromConfig(initialPort: Int) =
-    extractToxiproxyServersAndPorts().mapIndexed { i, (proxyname, serverVar, portVar, enabled) ->
-        val upstreamServer = fields["config/$serverVar"]?.value<String>()
-        val upstreamPort = fields["config/$portVar"]?.value<String>()
-        ToxiproxyConfig(
-            name = proxyname,
-            listen = "0.0.0.0:${initialPort + i}",
-            upstream = "$upstreamServer:$upstreamPort",
-            enabled = enabled
-        )
-    }
+        }
+}
 
 internal fun AuroraDeploymentSpec.databasesFromSecrets(
     initialPort: Int,
     databaseSchemaProvisioner: DatabaseSchemaProvisioner,
     userDetailsProvider: UserDetailsProvider
-): Pair<List<ToxiproxyConfig>, Map<String, Int>> = findDatabases(this)
-    .filter {
-        fields["toxiproxy/database"]?.value?.booleanValue() == true ||
-            fields["toxiproxy/database/" + it.name + "/enabled"]?.value?.booleanValue() == true
-    }
-    .createSchemaRequests(userDetailsProvider, this)
-    .associateWith { databaseSchemaProvisioner.findSchema(it) }
-    .filterNot { it.value == null }
-    .toList()
-    .mapIndexed { i, (request, schema) ->
-        val dbName = (request as SchemaForAppRequest).labels["name"]
-        val proxyname = fields["toxiproxy/database/$dbName/proxyname"]?.value<String>() ?: "database_${schema!!.id}"
-        val enabled = fields["toxiproxy/database/$dbName/initialEnabledState"]?.value<Boolean>() ?: true
+): Pair<List<ToxiproxyConfig>, Map<String, Int>> {
+    val schemas = findDatabases(this)
+        .filter {
+            if (this.isSimplifiedConfig(ToxiproxyField.database)) {
+                this[ToxiproxyField.database]
+            } else if (this.isSimplifiedConfig("${ToxiproxyField.database}/${it.name}")) {
+                this["${ToxiproxyField.database}/${it.name}"]
+            } else {
+                this["${ToxiproxyField.database}/${it.name}/enabled"]
+            }
+        }
+        .createSchemaRequests(userDetailsProvider, this)
+        .mapNotNull { request ->
+            val schema = databaseSchemaProvisioner.findSchema(request)
+
+            if (schema != null) {
+                request to schema
+            } else null
+        }
```

Comment:
Hvorfor ikke bruke `associateWith` her?

Reply:
vi kan gjøre det, men det jeg så var at det først ble gjort om til en map for så å bli konvertert til en list of pairs. Dvs en transformasjon ekstra. Jeg så ikke helt vitsen med det. Vi klarer å filtrere ut null for schema i tillegg her. Det går ann med map også, men usikker på hvilken verdi det gir å ha en slik mellom transformasjon. 
Hva tenker du?